mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react';
+import { PanelBody, Spinner } from '@wordpress/components';
+import { dispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { uniq } from 'lodash';
+import { storeName } from '../../../../../editor/store';
+import { MailPoetAjax } from '../../../../../../ajax';
+import { RawSegment, Segment } from './segment';
+import { PlainBodyTitle } from '../../../../../editor/components/panel';
+import {
+  FormTokenItem,
+  FormTokenField,
+} from '../../../components/form-token-field';
+
+export function ListPanel(): JSX.Element {
+  const { selectedStep } = useSelect(
+    (select) => ({
+      selectedStep: select(storeName).getSelectedStep(),
+    }),
+    [],
+  );
+
+  const [segments, setSegments] = useState<FormTokenItem[]>([]);
+
+  const anyValue = {
+    id: 0,
+    name: __('Any list', 'mailpoet'),
+  };
+
+  useEffect(() => {
+    const getData = async () => {
+      const data = await MailPoetAjax.post({
+        api_version: window.mailpoet_api_version,
+        endpoint: 'segments',
+        action: 'listing',
+      });
+      if (data?.data) {
+        const loadedSegments: FormTokenItem[] = data.data
+          .map((segment: RawSegment) => {
+            const sanitizedSegment: Segment = {
+              ...segment,
+              id: parseInt(segment.id, 10),
+            };
+            return sanitizedSegment;
+          })
+          .filter(
+            (segment: Segment): boolean =>
+              !['wp_users', 'woocommerce_users'].includes(segment.type),
+          );
+        setSegments(loadedSegments);
+      }
+    };
+
+    void getData();
+  }, []);
+
+  const rawSelected = selectedStep.args?.segment_ids
+    ? (selectedStep.args.segment_ids as number[])
+    : [];
+  const selected = segments.filter((segment): boolean =>
+    rawSelected.includes(segment.id as number),
+  );
+  if (!selected.length) {
+    selected.push(anyValue);
```

Comment:
As mentioned [above](https://github.com/mailpoet/mailpoet/pull/4371/commits/f4bde0a02b411648bace63f16310137e781d1863#diff-c993f976a5f0c987562cdc6ff34526cbe7d3d6fcc780207fbb204dd172528237R30), effects should never be done in a React component body, and it's often better to avoid mutability.

Reply:
Problem outdated thanks to [710190b](https://github.com/mailpoet/mailpoet/pull/4371/commits/710190bd68357719381a9d577bc886a495f5a367)